Feature/web/Label Page 구현 ( 추가 Event, CSS 작업 필요)

TwentyFiveSeven opened the pull request:
Label Page를 위한 컴포넌트들을 구현하였고, Router, Reducer등의 일관성있는 사용을 위한 코드를 작성하였습니다.
리뷰 후에 변경이 필요한 부분은 코멘트로 남겨주시면 감사하겠습니다 !

TwentyFiveSeven committed:
feature : webpack 추가 설정 및 package 설정

프로젝트를 진행하면서 style, css를 적용하기위해 loader를 다운받고
webpack에 추가 설정을 해주었습니다.

TwentyFiveSeven committed:
feature : React Router 설정

SPA에서 페이지 이동을 보여주기위해 Route와 Link를 사용하여 페이지가
이동하는 것처럼 보이게 하였습니다.

TwentyFiveSeven committed:
feature : 페이지 렌더링 템플릿

컴포넌트들로 구성된 페이지를 일관성 있게 렌더링하기 위해 완성된 페이지를
받아서 헤더, 푸터 등을 적용할 수 있는 템플릿을 만들었습니다

TwentyFiveSeven committed:
feature : 상태관리를 위한 reducer directory 생성

react에서 re-rendering 을 위해 상태를 변화시켜줘야합니다. 때문에
변화되는 상태를 관리하기위해서 Reducer를 만들어 사용하였습니다.

TwentyFiveSeven committed:
feature : Label Page의 root 구현

Label Page를 구현하기위해서 가장 상위에서 State, Context,  통신, 이벤트
등을 관리하는 함수를 구현하였습니다.

TwentyFiveSeven committed:
feature : 상단 Link 버튼들 구현

Label Page 가장 상단을 구현하기위해 Label, Milestone Page로 이동하고, 새로운 Label을
만들 수 있는 버튼을 구현하였습니다.

TwentyFiveSeven committed:
feature : Label의 수를 보여주는 컴포넌트 구현

라벨의 수를 보여주기위해 useContext로 상위 컴포넌트로부터 state를 받아와
개수를 출력해줬습니다

TwentyFiveSeven committed:
feature : LabelList, Label 컴포넌트 구현

LabelList 컴포넌트에서 상위 컴포넌트로부터 useContext를 사용하여 state를
받아와 state에 맞는 Label 컴포넌트들을 생성해줬습니다. Label
컴포넌트에서는 상위 컴포넌트의 함수를 받아 실행하는 이벤트핸들러를
구현하여 onClick에 걸어줬습니다

sunkest commented on Front-End/src/app.js:
```diff
@@ -1,11 +1,15 @@
+import { BrowserRouter, Route, Switch } from "react-router-dom";
+import * as Pages from "./pages";
 
 export default class App extends React.Component {
```
class 말고 함수형으로 쓰는게 요구사항에 더 적합한거 같습니다

sunkest replied:
나머지는 다 좋아요ㅎㅎ👍👍

sunkest replied:
아 그리고 다른 파일들에 function 형태로 함수 쓰셨는데 저는 arrow function을 더 선호하긴 합니다만 이건 스타일 차이니까 크게 상관없을 것 같아서 서로 하던대로 하면 될 것 같아요

NukeStorm replied:
코드 잘짜셔서 리액트 공부에 도움이 많이 되고 있습니다.
코드 참고해서 지금 구현하고 있는 컴포넌트 완성해보도록 하겠습니다

NukeStorm replied:
코드 잘짜셔서 리액트 공부에 도움이 많이 되고 있습니다.
코드 참고해서 지금 구현하고 있는 컴포넌트 완성해보도록 하겠습니다

TwentyFiveSeven replied:
넵넵 ! 화살표함수로 맞추고 ! 저부분은 .... 초기 코드를 쓰다보니 클래스였다는걸 알지 못했네요.. ㅎㅎㅎㅎㅎ 수정하겠습니다 !

TwentyFiveSeven committed:
fix : EventHandler 수정

push 또는 change가 완료 된 컴포넌트를 hidden으로 바꿔줬습니다.

TwentyFiveSeven committed:
fix : Label 컴포넌트에 key값 추가

반복되는 컴포넌트를 구현할 때는 구별할 수 있는 key 값이 필요하기 때문에
Label data의 id 값으로 Key값을 지정해줬습니다.

TwentyFiveSeven committed:
feature : 입력 이벤트와 Cancel 버튼 초기화 구현

사용자가 name을 입력하거나 변경하면 그에 맞게 변경되야하고 Cancel 버튼을
누르면 값들이 다시 돌아와야하기 때문에 state들을 사용하여
구현하였습니다.

TwentyFiveSeven committed:
fix : 클래스를 함수형으로 변환

초기 코드로 클래스로 선언되어있어 놓치고 넘어간 부분을 수정합니다.

TwentyFiveSeven committed:
feature : Label List Delete 액션 추가

라벨 리스트에서 라벨을 삭제하기 위해서 action의 type으로 delete 값을
받고 index값을 받아 해당 라벨을 삭제합니다.

TwentyFiveSeven committed:
feature : 상태 변경을 위해 하위 컴포넌트에게 넘겨줄 함수 구현

하위 컴포넌트에게 상태 변경을 위한 함수를 넘겨주기위해 useContext 를
사용하여 하위 컴포넌트에게 상태를 변경할 수 있는 함수를 넘겨주었습니다.

TwentyFiveSeven committed:
feature : Style을 위한 컴포넌트 구조 변경

Github 홈페이지와 유사한 CSS를 적용하기위해 추가적으로 Div 태그를 통해
컴포넌트들을 묶어주는 작업을 진행했습니다.

TwentyFiveSeven committed:
feature : 상태를 활용하기위한 Hooks사용

Event를 통해 값을 변경할 때 DOM API사용하는 번거로움을 줄이기 위해
Hooks를 통해 상태값을 활용하여 쉽게 데이터값을 바꿨습니다.
